src/main/java/io/github/moulberry/notenoughupdates/util/Utils.java

```diff
@@ -1470,4 +1476,15 @@ public static boolean isWithinRect(int x, int y, int left, int top, int width, i
         return left <= x && x <= left + width &&
                 top <= y && y <= top + height;
     }
+
+    public static <T, U> List<U> runningReduce(Iterable<T> values, U initial, BiFunction<T, U, U> reducer) {
```

Comment:
Why is this here? it is not used anywhere but was added in the PR?

Reply:
Right, i used that in an earlier iteration of the pet xp calculation, will remove.